CDCgov/prime-reportstream
# 10838: elims message part 4

This PR enables support for mapping AOE OBX segments in the SPECIMEN group into ServiceRequest.supportingInfo and then back to HL7

Test Steps:
Using both sample messages in the linked issue
1. Update the `dev.DEV_ELIMS` receiver to be active
2. Send the message in the linked issue to  `ignore.ignore-elr-elims` sender
3. Verify that the only discrepancies between the two are noted in the spreadsheet
https://cdc.sharepoint.com/:x:/r/teams/ReportStream/Shared%20Documents/ELIMS-ELR/[47_04608717_11184_mega_specimen.xlsx](https://cdc.sharepoint.com/:x:/r/teams/ReportStream/Shared%20Documents/ELIMS-ELR/47_04608717_11184_mega_specimen.xlsx?d=w7651af5fc4944146a679a0a961f0c526&csf=1&web=1&e=D97aMQ)?d=w7651af5fc4944146a679a0a961f0c526&csf=1&web=1&e=D97aMQ

## Changes
- Adds mappings for SPECIMEN.OBX -> ServiceRequest.supportingInfo
- Adds mapping for ServiceRequest.supportingInfo -> SPECIMEN.OBX
- Fixes an issue in the FHIR -> HL7 library that didn't support multiple elements appearing in the schema twice

## Checklist

### Testing

### Process

## Linked Issues
- #10838 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #10839
